index.js

```diff
@@ -1,33 +1,81 @@
-const paginationEmbed = async (msg, pages, emojiList = ['⏪', '⏩'], timeout = 120000) => {
+/**
+ *
+ * @param {MessageEmbed[]} pages
+ * @param {EmojiIdentifierResolvable[]} emojiList
+ * @param {number} currentPageIndex
+ * @param {MessageReaction} reaction
+ * @returns {number} - the new page index.
+ */
+const defaultPageResolver = async (_, pages, emojiList, currentPageIndex, reaction) => {
+	let newPage = currentPageIndex;
+	switch (reaction.emoji.name) {
+		case emojiList[0]:
+			newPage = currentPageIndex > 0 ? currentPageIndex - 1 : pages.length - 1;
+			break;
+		case emojiList[1]:
+			newPage = currentPageIndex + 1 < pages.length ? currentPageIndex + 1 : 0;
+			break;
+		default:
+			return currentPageIndex;
+	}
+	return newPage;
+};
+
+const defaultFooterResolver = (currentPageIndex, pagesLength) =>`Page ${currentPageIndex + 1} / ${pagesLength}`;
+
+/**
+ *
+ * @param {Message} msg - the message
+ * @param {MessageEmbed[]} pages - array of message embeds to use as each page.
+ * @param {PaginationOptions} paginationOptions - exposes collector options, provides customization.
+ */
+const paginationEmbed = async (
+										msg, pages,
+										{
+											deleteOnEnd = false, useUtil = false,
+											emojiList = ['⏪', '⏩'],
+											collectorFilter = null,
+											footerResolver = defaultFooterResolver,
+											pageResolver = defaultPageResolver,
+											timeout = 120000,
+											...rest
+										} = {}) => {
 	if (!msg && !msg.channel) throw new Error('Channel is inaccessible.');
 	if (!pages) throw new Error('Pages are not given.');
-	if (emojiList.length !== 2) throw new Error('Need two emojis.');
 	let page = 0;
-	const curPage = await msg.channel.send(pages[page].setFooter(`Page ${page + 1} / ${pages.length}`));
-	for (const emoji of emojiList) await curPage.react(emoji);
+	pages[page].setFooter(footerResolver(page, pages.length));
+	const curPage = useUtil ? await msg.util.reply(pages[page]) : await msg.channel.send(pages[page]);
 	const reactionCollector = curPage.createReactionCollector(
-		(reaction, user) => emojiList.includes(reaction.emoji.name) && !user.bot,
-		{ time: timeout }
+		(reaction, user) => emojiList.includes(reaction.emoji.name) && !user.bot
+							&& (collectorFilter ? collectorFilter(reaction, user, emojiList) : true),
```

Comment:
Ditto here with the borked indentation.

Reply:
Also, `collectorFilter` should both default to `() => true` and not `null`, and it should be awaited, since you type it as returning either `boolean` or `Promise<boolean>`.